src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -0,0 +1,401 @@
+                    logActivity name stat.count stat.ms
+                    rows <- rows + 1
+            // Yes, there's a minor race here between the use of the values and the reset
+            let duration = Stats.LogSink.Restart()
+            if rows > 1 then logActivity "TOTAL" totalCount totalMs
+            let measures : (string * (TimeSpan -> float)) list = [ "s", fun x -> x.TotalSeconds(*; "m", fun x -> x.TotalMinutes; "h", fun x -> x.TotalHours*) ]
+            let logPeriodicRate name count = log.Information("rp{name} {count:n0}", name, count)
+            for uom, f in measures do let d = f duration in if d <> 0. then logPeriodicRate uom (float totalCount/d |> int64)
+
+
+[<RequireQualifiedAccess; NoEquality; NoComparison>]
+type EsSyncResult = Written of int64 | ConflictUnknown
+
+module private Write =
+    /// Yields `EsSyncResult.Written` or `EsSyncResult.Conflict` to signify WrongExpectedVersion
+    let private writeEventsAsync (log : ILogger) (conn : IEventStoreConnection) (streamName : string) (version : int64) (events : EventData[])
+        : Async<EsSyncResult> = async {
+        try let! wr = conn.WriteMessages(streamName, events, version, Async.DefaultCancellationToken) |> Async.AwaitTaskCorrect
+            return EsSyncResult.Written (version + int64 events.Length)
+        with WrongExpectedVersion as ex ->
+            log.Information(ex, "SqlEs TrySync WrongExpectedVersionException writing {EventTypes}, expected {ExpectedVersion}",
+                [| for x in events -> x.EventType |], version)
+            return EsSyncResult.ConflictUnknown }
+    let eventDataBytes events =
+        let eventDataLen (x : EventData) = match x.Data, x.Meta with Log.BlobLen bytes, Log.BlobLen metaBytes -> bytes + metaBytes
+        events |> Array.sumBy eventDataLen
+    let private writeEventsLogged (conn : IEventStoreConnection) (streamName : string) (version : int64) (events : EventData[]) (log : ILogger)
+        : Async<EsSyncResult> = async {
+        let log = if (not << log.IsEnabled) Events.LogEventLevel.Debug then log else log |> Log.propEventData "Json" events
+        let bytes, count = eventDataBytes events, events.Length
+        let log = log |> Log.prop "bytes" bytes
+        let writeLog = log |> Log.prop "stream" streamName |> Log.prop "expectedVersion" version |> Log.prop "count" count
+        let! t, result = writeEventsAsync writeLog conn streamName version events |> Stopwatch.Time
+        let reqMetric : Log.Measurement = { stream = streamName; interval = t; bytes = bytes; count = count}
+        let resultLog, evt =
+            match result, reqMetric with
+            | EsSyncResult.ConflictUnknown, m ->
+                log, Log.WriteConflict m
+            | EsSyncResult.Written x, m ->
+                log |> Log.prop "currentPosition" x, Log.WriteSuccess m
+        (resultLog |> Log.event evt).Information("MsgdbEs{action:l} count={count} conflict={conflict}",
```

Comment:
`MsgdbEs` just needs to be a relatively unique strong to grep in log output. Maybe `Msgdb` is enough? (just an idea...)